### PR TITLE
[patch] Support for CP4D 4.8 in Gitops and ArgoCD

### PR DESCRIPTION
**Pull request by caroazad**
Add file template and configure CP4D 4.8 to deploy with argoCD. Supporting 4.8.0 and 4.8.1
Changes tested and passed as shown below in this image (image showing a run in argocd for cp4d 4.8.0)

Screenshot 2024-06-01 at 1 32 04 AM

### CODE REVIEW

File: image/cli/mascli/functions/gitops_cp4d [hunk 1]
```diff
@@ -13,6 +13,7 @@ GitOps Configuration:
   -d, --dir ${COLOR_YELLOW}GITOPS_WORKING_DIR${TEXT_RESET}           Directory for GitOps repository
   -g, --gitops-version ${COLOR_YELLOW}GITOPS_VERSION${TEXT_RESET}    Version of ibm-mas/gitops to use
   -a, --account-id ${COLOR_YELLOW}ACCOUNT_ID${TEXT_RESET}            Account name that the cluster belongs to
+  -r, --region-id ${COLOR_YELLOW}CLUSTER_ID${TEXT_RESET}             Region ID
```
Comment (whitfiea) [inline image/cli/mascli/functions/gitops_cp4d]: CLUSTER_ID should be REGION_ID

File: image/cli/mascli/functions/gitops_cp4d [hunk 2]
```diff
@@ -52,6 +53,9 @@ function gitops_cp4d_noninteractive() {
   GIT_COMMIT_MSG="gitops-cp4d commit"
   export GITOPS_VERSION=${GITOPS_VERSION:-poc}
 
+  export REGION=${REGION:-${SM_AWS_REGION}}
```
Comment (whitfiea) [inline image/cli/mascli/functions/gitops_cp4d]: can this be:
```
  export REGION_ID=${REGION_ID:-${SM_AWS_REGION}}
```

File: image/cli/mascli/functions/gitops_cp4d [hunk 3]
```diff
@@ -70,6 +74,9 @@ function gitops_cp4d_noninteractive() {
       -c|--cluster-id)
         export CLUSTER_ID=$1 && shift
         ;;
+      -r|--region)
+        export REGION=$1 && shift
```
Comment (whitfiea) [inline image/cli/mascli/functions/gitops_cp4d]: REGION_ID and --region-id

File: image/cli/mascli/functions/gitops_cp4d [hunk 4]
```diff
@@ -147,8 +154,9 @@ function gitops_cp4d_noninteractive() {
   [[ -z "$GITOPS_WORKING_DIR" ]] && gitops_cp4d_help "GITOPS_WORKING_DIR is not set"
   [[ -z "$ACCOUNT_ID" ]] && gitops_cp4d_help "ACCOUNT_ID is not set"
   [[ -z "$CLUSTER_ID" ]] && gitops_cp4d_help "CLUSTER_ID is not set"
+  [[ -z "$REGION" ]] && ggitops_cp4d_help "REGION is not set"
```
Comment (whitfiea) [inline image/cli/mascli/functions/gitops_cp4d]: should be REGION_ID

File: image/cli/mascli/functions/gitops_cp4d [hunk 5]
```diff
@@ -238,21 +245,68 @@ function gitops_cp4d() {
...
+  export SECRET_KEY_CPD_PLATFORM_CHANNEL=${ACCOUNT_ID}${SECRETS_KEY_SEPERATOR}${CLUSTER_ID}${SECRETS_KEY_SEPERATOR}${MAS_INSTANCE_ID}${SECRETS_KEY_SEPERATOR}cpd_platform_channel#cpd_platform_channel
+
+
+  # Default settings
+  export CPD_CS_CONTROL_NAMESPACE="cs-control"
+  export CPD_ADMIN_LOGIN_SA="mas-cpd-cluster-admin-sa"
+  export CPD_PLATFORM_CR_NAME="ibmcpd-cr"
+  export CPD_IAM_INTEGRATION=false
+  export CPFS_SIZE="small"
+  export CPD_SCALE_CONFIG="medium"
```
Comment (whitfiea) [inline image/cli/mascli/functions/gitops_cp4d]: We would want to have these last two configurable:
```
  export CPFS_SIZE="small"
  export CPD_SCALE_CONFIG="medium"
```

So have the defaults but let those be overwritten if are passed in. these would also need to be exposed on the task/pipeline

**Review by whitfiea — CHANGES_REQUESTED** [review]
How do we set the catalogsource versions in this file: https://github.com/ibm-mas/gitops/blob/poc/instance-applications/080-ibm-cp4d/templates/01-ibm-cp4d_CatalogSources.yaml#L16

**Comment by caroazad** [comment]
> How do we set the catalogsource versions in this file: https://github.com/ibm-mas/gitops/blob/poc/instance-applications/080-ibm-cp4d/templates/01-ibm-cp4d_CatalogSources.yaml#L16

I did not import them because it was only one set and only runs when cp4d version is 4.8.0

File: tekton/src/pipelines/gitops/gitops-mas-apps.yml.j2 [hunk 6]
```diff
@@ -135,6 +135,10 @@ spec:
       type: string
     - name: cp4d_action
       type: string
+    - name: cpfs_size
+      type: string
+    - name: cpd_scale_config
+      type: string
```
Comment (whitfiea) [inline tekton/src/pipelines/gitops/gitops-mas-apps.yml.j2:142]: We should set `default: ""` for these as these would be optional, and we need a default value

**Review by whitfiea — CHANGES_REQUESTED** [review]

**Review by whitfiea — APPROVED** [review]

**Pull request merged**